de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/versioningbackend/GitVersioningBackendTest.java

```diff
@@ -135,21 +135,28 @@ protected void testCommitProjectAssert() {
 	 * @return The Commit message as string.
 	 */
 	protected String getCommitMessageFor(String repoName) {
-		// Assert that commit message arrived in Local Repository
-		bot.viewByTitle("Git Repositories").show();
-		getTreeNodeContaining(repoName).select();
-		
-		// Open the history view and get the entry from the commit
-		SWTBotView historyView = bot.view(WithTitle.withTitle(StringContains.containsString("History")));
-		historyView.show();
-		SWTBotTable historyTable =  historyView.bot().table();
-		SWTBotTableItem historyTableItem0 = historyTable.getTableItem(0);
-	
-		// Retrieve the commit message
-		String commitMessage = historyTableItem0.getText(1);
-		return commitMessage;
+	    // Assert that commit message arrived in Local Repository
+	    bot.viewByTitle("Git Repositories").show();
+	    getTreeNodeContaining(repoName).select();
+
+	    // Open the history view and get the entry from the commit
+	    SWTBotView historyView = bot.view(WithTitle.withTitle(StringContains.containsString("History")));
+	    historyView.show();
+	    SWTBotTable historyTable =  historyView.bot().table();
+	    SWTBotTableItem historyTableItem0 = historyTable.getTableItem(0);
+
+	    // Retrieve the commit message
+	    String commitMessage = historyTableItem0.getText(1);
+	    
+	    // If the commit message contains the phrase "Backend Local Commit Before Pull", extract the actual commit message
+	    if (commitMessage.contains("Backend Local Commit Before Pull")) {
+	        commitMessage = commitMessage.replace("Backend Local Commit Before Pull: ", "");

```

Comment:
That is right, i removed it.

Reply:
Instead of changing (replacing "Backend Local Commit Before Pull: ") the string inside the test, I think the cleaner version would be to change the string in the assert function in the test